xla/service/gpu/triton_tiling_propagation.cc

```diff
@@ -1079,10 +1079,10 @@ GetPropagatedDimOrdersAndRequirementsIfProfitablyFusible(
       if (i == *src_operand_index) {
         continue;
       }
-      // Currently only broadcasts of scalar constants or parameters
-      // are accepted as other inputs of non-unary operations
-      // in the output fusion.
-      if (hlo_query::IsBroadcastOfScalarConstant(*operand) ||
+      // Currently only broadcasts of scalars or parameters are accepted as
+      // other inputs of non-unary operations in the output fusion.
+      if ((operand->opcode() == HloOpcode::kBroadcast &&
+           ShapeUtil::IsEffectiveScalar(operand->operand(0)->shape())) ||
```

Comment:
Are you sure that this is supported in codegen? I seemed to recall broadcast failures for non-0d shapes.

Please add a corresponding emitter test.

Reply:
Good catch thanks - I tried to find some example test cases but I think there are currently only broadcast of scalars. I might have a wrong impression that we need to enable effective-scalars. 
Changed to fusing only scalar instructions. Also added test in emitter test.